WorkflowCombine/Sources/PublisherWorkflow.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(Combine) && swift(>=5.1)
+
+    import Combine
+    import Foundation
+    import Workflow
+
+    @available(iOS 13.0, macOS 10.15, *)
+
+    extension AnyPublisher: AnyWorkflowConvertible where Failure == Never {
```

Comment:
This is a workaround to the fact Swift limits you from being able to extend a protocol with a new inheritance, for example:
```
extension Publisher: AnyWorkflowConvertible {}
```
Which in turn disallows a holistic inheritance of all publishers (without extending each individual one) to `AnyWorkflowConvertible`. The current implementation only extends `AnyPublisher`, which forces developers to have to type-erase their publishers before being able to render it.